apps/server/src/modules/provisioning/strategy/sanis/response/sanis.response.ts

```diff
@@ -1,10 +1,20 @@
+import { Type } from 'class-transformer';
+import { ArrayMinSize, IsArray, IsString, ValidateNested } from 'class-validator';
 import { SanisPersonResponse } from './sanis-person-response';
 import { SanisPersonenkontextResponse } from './sanis-personenkontext-response';
+import { SanisResponseValidationGroups } from './sanis-response-validation-groups';
 
-export interface SanisResponse {
-	pid: string;
+export class SanisResponse {
+	@IsString({ groups: [SanisResponseValidationGroups.USER] })
+	pid!: string;
```

Comment:
The non null assertions here are used in the same way they are used on the controller dtos. Since we get a plain object over the api, it has to be transformed to a class in order for the annotations to work/exist on the object. Due to this we cannot use a constructor that is not the default constructor, because the transformer would not work anymore. This is the same way that Nest does it internally.

The transformer fills out all fields on the object that it finds in the plain object. Then we run the validation, that reassures us, that all fields on the object are valid just like we defined it.

Now, the compiler would complain when we didn't use a non null assertion, because fields on a class have to be initialized (or they are optional) to assure type safety. But we use a different way of initialization then the compiler knows and we can still guaranty type safety (as long as the object is used in the correct context).

Thats why we can use non null assertions here. 🙂

Reply:
Ah okay i see, thanks for explanation :) 
So default constructor is used for transformation so we cant use a constructor but we have to initialize those properties -> so we need these operators